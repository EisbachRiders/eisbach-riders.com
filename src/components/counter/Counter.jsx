import React from 'react'
import { makeStyles } from '@material-ui/styles'
import IconButton from '@material-ui/core/IconButton'
import RemoveIcon from '@material-ui/icons/Remove'
import AddIcon from '@material-ui/icons/Add'
import InputBase from '@material-ui/core/InputBase'
import InputLabel from '@material-ui/core/InputLabel'

const useStyles = makeStyles(theme => ({
  container: {
    display: 'flex',
  },
  input: {
    width: 32,
  },
  inputProp: {
    textAlign: 'right',
  },
}))

function Counter({ handleRemove, handleAdd, value }) {
  const classes = useStyles()

  return (
    <div className={classes.container}>
      <IconButton className={classes.iconButton} onClick={() => handleRemove()}>
        <RemoveIcon />
      </IconButton>
      <form className={classes.container} noValidate autoComplete="off">
        <InputLabel htmlFor="counter" className="visually-hidden">
          Quantity
        </InputLabel>
        <InputBase
          id="counter"
          readOnly
          className={classes.input}
          classes={{ input: classes.inputProp }}
          value={value}
          margin="none"
          type="number"
        />
      </form>
      <IconButton className={classes.iconButton} onClick={() => handleAdd()}>
        <AddIcon />
      </IconButton>
    </div>
  )
}

export default Counter
