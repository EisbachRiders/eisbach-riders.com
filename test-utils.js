import React from 'react'
import { render } from '@testing-library/react'
import { I18nextProvider } from 'react-i18next'
import i18n from './src/i18n/i18n'
import { ThemeProvider } from '@material-ui/styles'
import theme from './src/theme'

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
    </ThemeProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
