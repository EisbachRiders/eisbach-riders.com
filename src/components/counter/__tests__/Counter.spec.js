import React from 'react'
import { render } from '../../../../test-utils'
import Counter from '../Counter'

describe('Counter', () => {
  it('renders correctly', () => {
    const component = render(<Counter />)
    expect(component).toMatchSnapshot()
  })
})
