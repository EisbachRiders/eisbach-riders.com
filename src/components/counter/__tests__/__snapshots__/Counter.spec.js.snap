// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-container-1"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13H5v-2h14v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <form
          autocomplete="off"
          class="makeStyles-container-1"
          novalidate=""
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root visually-hidden MuiInputLabel-animated"
            for="counter"
          >
            Quantity
          </label>
          <div
            class="MuiInputBase-root makeStyles-input-2"
          >
            <input
              class="MuiInputBase-input makeStyles-inputProp-3"
              id="counter"
              readonly=""
              type="number"
              value=""
            />
          </div>
        </form>
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-container-1"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13H5v-2h14v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <form
        autocomplete="off"
        class="makeStyles-container-1"
        novalidate=""
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root visually-hidden MuiInputLabel-animated"
          for="counter"
        >
          Quantity
        </label>
        <div
          class="MuiInputBase-root makeStyles-input-2"
        >
          <input
            class="MuiInputBase-input makeStyles-inputProp-3"
            id="counter"
            readonly=""
            type="number"
            value=""
          />
        </div>
      </form>
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
